runtime/vam/expr/function/network_of.go

```diff
@@ -0,0 +1,218 @@
+package function
+
+import (
+	"net/netip"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/vector"
+)
+
+// https://github.com/brimdata/zed/blob/main/docs/language/functions.md#network_of
+type NetworkOf struct {
+	zctx *zed.Context
+}
+
+func (n *NetworkOf) Call(args ...vector.Any) vector.Any {
+	args = underAll(args)
+	ipvec := args[0]
+	if ipvec.Type().ID() != zed.IDIP {
+		return vector.NewWrappedError(n.zctx, "network_of: not an IP", ipvec)
+	}
+	if len(args) == 1 {
+		return n.singleIP(ipvec)
+	}
+	maskvec := args[1]
+	switch id := maskvec.Type().ID(); {
+	case id == zed.IDIP:
+		return n.ipMask(ipvec, maskvec)
+	case zed.IsInteger(id):
+		return n.intMask(ipvec, maskvec)
+	default:
+		return vector.NewWrappedError(n.zctx, "network_of: bad arg for CIDR mask", maskvec)
+	}
+}
+
+func (n *NetworkOf) singleIP(vec vector.Any) vector.Any {
+	if c, ok := vec.(*vector.Const); ok {
+		ip, _ := vector.IPValue(vec, 0)
+		if !ip.Is4() {
+			return errNotIP4(n.zctx, vec)
+		}
+		net := netip.PrefixFrom(ip, bitsFromIP(ip.As4())).Masked()
+		return vector.NewConst(zed.NewNet(net), c.Len(), c.Nulls)
+	}
+	var errs []uint32
+	var nets []netip.Prefix
+	switch vec := vec.(type) {
+	case *vector.IP:
+		nets, errs = n.singleIPLoop(vec, nil)
+	case *vector.View:
+		nets, errs = n.singleIPLoop(vec.Any.(*vector.IP), vector.NewSelection(vec))
+	case *vector.Dict:
+		var derrs []uint32
+		nets, derrs = n.singleIPLoop(vec.Any.(*vector.IP), nil)
+		if len(derrs) > 0 {
+			vec, errs = vec.DeleteEntries(derrs...)
+		}
+	}
+	out := vector.NewNet(nets, nil)
+	if len(errs) > 0 {
+		return vector.Combine(out, errs, errNotIP4(n.zctx, vector.NewView(errs, vec)))
+	}
+	return out
+}
+
+func (n *NetworkOf) singleIPLoop(vec *vector.IP, sel *vector.Selection) ([]netip.Prefix, []uint32) {
+	size := vec.Len()
+	var nets []netip.Prefix
+	var errs []uint32
+	for i := uint32(0); i < size; i++ {
```

Comment:
```suggestion
	var nets []netip.Prefix
	var errs []uint32
	for i := range vec.Len() {
```

(This evaluates `vec.Len()` once.)